### PR TITLE
Clean up Python code generator

### PR DESCRIPTION
**Pull request by clinssen**
Just a little cleanup of the Python code generator; added some first steps for synapse support (but not functional yet, hence no added tests).

### CODE REVIEW

File: pynestml/codegeneration/resources_python_standalone/point_neuron/@SYNAPSE_NAME@.py.jinja2 [hunk 1]
```diff
@@ -0,0 +1,396 @@
...
+import math
+from math import *
+import numpy as np
+import scipy
+import scipy.integrate
+
+from .synapse import Synapse
+from .utils import steps
+
+DEBUG = 1
```
Comment (pnbabu) [inline pynestml/codegeneration/resources_python_standalone/point_neuron/@SYNAPSE_NAME@.py.jinja2:59]: Please remove this if it's not used.

File: pynestml/codegeneration/resources_python_standalone/point_neuron/@SYNAPSE_NAME@.py.jinja2 [hunk 2]
```diff
@@ -0,0 +1,396 @@
...
+
+{%-     include "directives_py/AnalyticIntegrationStep_end.jinja2" %}
+{%- endfilter %}
+{%- endfor %}
+{%- endif %}
+
+  def step(self, origin: float, timestep: float) -> None:
+    __timestep: float = timestep   # do not remove, this is necessary for the timestep() function
+
+    assert False, "Synapsese are not yet supported for the Python-standalone code generation target."
```
Comment (pnbabu) [inline pynestml/codegeneration/resources_python_standalone/point_neuron/@SYNAPSE_NAME@.py.jinja2:290]:
```suggestion
    assert False, "Synapses are not yet supported for the Python-standalone code generation target."
```

**Comment by clinssen** [comment]
Thanks for the review! Could you check if the latest updates address your comments?

**Review by pnbabu — APPROVED** [review]
Yes, thanks for the cleanup!

**Pull request merged**